gammapy/cube/fit.py

```diff
@@ -901,6 +901,404 @@ def to_image(self, spectrum=None, keepdims=True):
+
+        gti = GTI.create([] * u.s, [] * u.s, reference_time=reference_time)
+
+        mask_safe = np.zeros(geom.data_shape, dtype=bool)
+
+        return cls(
+            counts=maps["counts"],
+            counts_off=maps["counts_off"],
+            acceptance=maps["acceptance"],
+            acceptance_off=maps["acceptance_off"],
+            exposure=exposure,
+            psf=psf,
+            edisp=edisp,
+            gti=gti,
+            mask_safe=mask_safe,
+            name=name,
+            **kwargs,
+        )
+
+    @classmethod
+    def create(
+        cls,
+        geom,
+        geom_irf=None,
+        migra_axis=None,
+        rad_axis=None,
+        reference_time="2000-01-01",
+        name="",
+        **kwargs,
+    ):
+        """Creates a MapDataset object with zero filled maps
+
+        Parameters
+        ----------
+        geom: `~gammapy.maps.WcsGeom`
+            Reference target geometry in reco energy, used for counts and background maps
+        geom_irf: `~gammapy.maps.WcsGeom`
+            Reference image geometry in true energy, used for IRF maps.
+        migra_axis: `~gammapy.maps.MapAxis`
+            Migration axis for the energy dispersion map
+        rad_axis: `~gammapy.maps.MapAxis`
+            Rad axis for the psf map
+        reference_time: `~astropy.time.Time`
+            the reference time to use in GTI definition
+        name : str
+            Name of the dataset.
+        """
+        geom_irf = geom_irf or geom.to_binsz(BINSZ_IRF)
+        migra_axis = migra_axis or MIGRA_AXIS_DEFAULT
+        rad_axis = rad_axis or RAD_AXIS_DEFAULT
+        energy_axis = geom_irf.get_axis_by_name("ENERGY")
+
+        geom_exposure = geom.to_image().to_cube([energy_axis])
+        geom_psf = geom_irf.to_image().to_cube([rad_axis, energy_axis])
+        geom_edisp = geom_irf.to_image().to_cube([migra_axis, energy_axis])
+
+        return cls.from_geoms(
+            geom=geom,
+            geom_exposure=geom_exposure,
+            geom_psf=geom_psf,
+            geom_edisp=geom_edisp,
+        )
+
+    def stack(self, other):
+        """Stack another MapDatasetOnOff in place.
+
+        The `acceptance` of the stacked dataset is normalized to 1, and the stacked `acceptance_off`
+        is scaled so that:
+
+        .. math::
+            \alpha_\text{stacked} = \frac{1}{a_\text{off}} = \frac{\alpha_1\text{OFF}_1
+            + \alpha_2\text{OFF}_2}{\text{OFF}_1 + OFF_2}
+
+        Parameters
+        ----------
+        other: `~gammapy.cube.MapDatasetOnOff`
+            Dataset to be stacked with this one.
+        """
+        super().stack(other)
+
+        if self.alpha and other.alpha and self.counts_off and other.counts_off:
+            self.acceptance_off.data = 1 / (
+                self.alpha.data * self.counts_off.data
+                + other.alpha.data * other.counts_off.data
+            )
+            self.acceptance.data = np.ones(self.data_shape)
```

Comment:
This looks incorrect indeed. 
- mask should be applied to compute the average `alpha`
- the average alpha has to be divided by the total OFF